tests/FSharp.Compiler.ComponentTests/ErrorMessages/TailCallAttribute.fs

```diff
@@ -1486,3 +1486,30 @@ namespace N
               Message =
                            "The TailCall attribute should only be applied to recursive functions." }
         ]
+
+    [<FSharp.Test.FactForNETCOREAPP>]
+    let ``Warn about self-defined attribute`` () = // is the analysis available for uers of older FSharp.Core versions
```

Comment:
Also, stupid question (I am not very strong at F# component tests) - how does this test verify the older version of F# core? It still requires "withLangVersionPreview".

Reply:
You are right, this test doesn't use an older `FSharp.Core` package (e.g. v6.0.7). As far as I know, we don't have infrastructure for such tests and the `withLangVersionPreview` doesn't affect the used `FSharp.Core` version.

The reason why I still think that this test is valid is the precedence that the locally defined Attribute takes in this test.
For example, if you rename the locally defined Attribute to `Foo` and adjust the comparison function in `TcGlobals.fs` accordingly, the test still passes.
To make this a bit more obvious, I removed the "withLangVersionPreview" from the test.

Reply:
Can you please make a separate example project inside `tests/projects` which uses locally built compiler and older FSharp.Core, and check that it works as expected?

Reply:
Sure, I added one and checked that it works as expected.

Reply:
Thanks, I'm slightly surprised that adding only `<DotnetFscCompilerPath>` was enough. It wasn't before.